File: app/src/main/java/com/ufkoku/ciphersharedpreferences/demo/dagger2/app_context/AppContextModule.java
package com.ufkoku.ciphersharedpreferences.demo.dagger2.app_context;

import android.content.Context;
import android.support.annotation.NonNull;

import dagger.Module;
import dagger.Provides;

@Module
public class AppContextModule {

    private Context app;

    public AppContextModule(Context app) {
        this.app = app;
    }

    @Provides
    @NonNull
    public Context provideAppContext() {
        return app;
    }

}

File: app/src/main/kotlin/com/ufkoku/ciphersharedpreferences/demo/dagger2/app_context/AppContextModule.kt
package com.ufkoku.ciphersharedpreferences.demo.dagger2.app_context

import android.content.Context

import dagger.Module
import dagger.Provides

@Module
class AppContextModule(private val app: Context) {

    @Provides
    fun provideAppContext(): Context {
        return app
    }

}
